dask/base.py

```diff
@@ -93,6 +93,18 @@ def persist(self, **kwargs):
         metadata, but now with the results fully computed or actively computing
         in the background.
 
+        The action of function differs significantly depending on the active
+        task scheduler.  If the task scheduler supports asynchronous computing,
+        such as is the case of the dask.distributed scheduler, then persist
+        will return *immediately* and the return value's task graph will
+        contain Dask Future objects.  However if the task scheduler only
+        supports blocking computation then the call to persist with *block*
```

Comment:
IIUC should be `will *block*`. PR ( https://github.com/dask/dask/pull/3305 ) should address this.